h2o-core/src/main/java/water/api/ModelBuilderHandler.java

```diff
@@ -38,11 +39,23 @@ public S handle(int version, Route route, Properties parms, String postBody) thr
 
     // User specified key, or make a default?
     String model_id = parms.getProperty("model_id");
+    String warningStr = null;
+    boolean modifiedModelID = false;
+    if ((model_id != null) && (model_id.contains("/"))) { // found / in model_id, replace with _ and set warning
+      String tempName = model_id;
+      model_id = model_id.replaceAll("/", "_");
+      warningStr = "Bad model_id: slash (/) found and replaced with _.  " + "Original model_id "+tempName +
+              " is now "+model_id+".";
+      Log.warn("model_id", warningStr);
+      modifiedModelID = true;
+    }
     Key<Model> key = doTrain ? (model_id==null ? ModelBuilder.defaultKey(algoName) : Key.<Model>make(model_id)) : null;
     // Default Job for just this training
-    Job job = doTrain ? new Job<>(key,ModelBuilder.javaName(algoURLName),algoName) : null;
```

Comment:
Can this be reverted back to the original like (43) and just add

if (job != null && warningStr != null)
   job.warn(warningStr);